ui/StatusQ/include/StatusQ/keychain.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <QObject>
+#include <QFuture>
+
+#ifdef __OBJC__
+#include <LocalAuthentication/LAContext.h>
+#else
+class LAContext;
+#endif
+
+class Keychain : public QObject {
+    Q_OBJECT
+
+    Q_PROPERTY(QString service READ service WRITE setService NOTIFY serviceChanged)
+    Q_PROPERTY(QString reason READ reason WRITE setReason NOTIFY reasonChanged)
+    Q_PROPERTY(bool loading READ loading NOTIFY loadingChanged)
+
+public:
+    explicit Keychain(QObject *parent = nullptr);
+    ~Keychain();
+
+    enum Status {
+        StatusSuccess = 0,
+        StatusNotSupported,
+        StatusGenericError,
```

Comment:
```suggestion
        StatusUnknownError,
```
Just a suggestion: that name seems more suitable here for me.

Reply:
I thought of this, but it's the error is actually _known_, that's why I went with _generic_.
But I don't insist, any approach works for me,

Reply:
@micieslak maybe `UnexpectedError` would fit? 
Kinda _known_, but not expected to happen 🤔

Reply:
Ok, let's keep it as it is, it's fine.